### PR TITLE
Utilities/tar: Open archive file before changing directory

### PR DESCRIPTION
**Pull request by supercomputer7**

### CODE REVIEW

File: Userland/Utilities/tar.cpp [hunk 1]
```diff
@@ -57,6 +57,20 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.add_positional_argument(paths, "Paths", "PATHS", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
+    auto program_name = LexicalPath::basename(arguments.strings[0]);
+
+    // NOTE: If the user run this program via the /bin/untar symlink,
+    // then emulate extraction.
+    if (program_name == "untar"sv) {
+        if (paths.size() > 1) {
+            warnln("exactly one file can be extracted with untar");
```
Comment (timschumi) [inline Userland/Utilities/tar.cpp:66]: Should probably be prefixed with `error:` or something like that.

File: Userland/Utilities/tar.cpp [hunk 2]
```diff
@@ -57,6 +57,20 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.add_positional_argument(paths, "Paths", "PATHS", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
+    auto program_name = LexicalPath::basename(arguments.strings[0]);
+
+    // NOTE: If the user run this program via the /bin/untar symlink,
```
Comment (timschumi) [inline Userland/Utilities/tar.cpp:62]: Typo: run*s* / ran

**Review by timschumi — CHANGES_REQUESTED** [review]
Not a blocker in itself (although if this is the case then I'd prefer not adding the symlink), but is `untar` a somewhat standardized command? I don't recall ever seeing a dedicated `untar` command anywhere.

**Comment by supercomputer7** [comment]
> Not a blocker in itself (although if this is the case then I'd prefer not adding the symlink), but is `untar` a somewhat standardized command? I don't recall ever seeing a dedicated `untar` command anywhere.

No, `untar` is not a standard command. But I don't see a reason to block this because of this. It simply was my own idea after seeing that there's `gunzip`, `unzip`, and `unzstd`... which all do the opposite operation of the original commands, so why not have something for `tar`? :)

**Comment by timschumi** [comment]
> > Not a blocker in itself (although if this is the case then I'd prefer not adding the symlink), but is `untar` a somewhat standardized command? I don't recall ever seeing a dedicated `untar` command anywhere.
> 
> No, `untar` is not a standard command. But I don't see a reason to block this because of this. It simply was my own idea after seeing that there's `gunzip`, `unzip`, and `unzstd` (on my host linux machine)... which all do the opposite operation of the original commands, so why not have something for `tar`? :)

The main reason we have these (at least as far as I am concerned) is for compatibility. All of these are existing commands that people who are familiar with the original implementations might try to use.

It probably isn't worthwhile to keep adding a bunch of symlinks and matching special cases (`un*`, `*cat`, etc.) to tools just to save two keystrokes at a time.

**Comment by supercomputer7** [comment]
> > > Not a blocker in itself (although if this is the case then I'd prefer not adding the symlink), but is `untar` a somewhat standardized command? I don't recall ever seeing a dedicated `untar` command anywhere.
> > 
> > 
> > No, `untar` is not a standard command. But I don't see a reason to block this because of this. It simply was my own idea after seeing that there's `gunzip`, `unzip`, and `unzstd` (on my host linux machine)... which all do the opposite operation of the original commands, so why not have something for `tar`? :)
> 
> The main reason we have these (at least as far as I am concerned) is for compatibility. All of these are existing commands that people (or scripts) who are familiar with the original implementations might try to use.
> 
> It probably isn't worthwhile to keep adding a bunch of symlinks and matching special cases (`un*`, `*cat`, etc.) to tools just to save two keystrokes at a time.

Hmm, that makes sense to me. I will drop the second commit then.

**Review by timschumi — APPROVED** [review]

**Pull request merged**